src/v/rpc/transport.cc

```diff
@@ -149,6 +149,9 @@ transport::make_response_handler(
 
     auto entry = std::make_unique<response_entry>();
 
+    // Resource units must be released when we notify caller with the result.
```

Comment:
Think we release them even earlier, right after dispatching send once the order is guaranteed (L307).

Reply:
fixed